apps/olat-api/src/index.ts

```diff
@@ -0,0 +1,488 @@
+const API_KEY = process.env.OLAT_API_KEY
+
+enum StatusCode {
+  SUCCESS = 200, // Request succeeded
+  BAD_REQUEST = 400, // Malformed request (e.g. missing/invalid parameter, invalid header)
+  UNAUTHORIZED = 401, // Invalid or missing API key
+  NOT_FOUND = 404, // Resource not found (optional, e.g. courseID not found)
+  UNSUPPORTED_MEDIA_TYPE = 415,
+  INTERNAL_SERVER_ERROR = 500,
+}
+
+interface ActivityType {
+  id: string
+  title: string
+  path: string
+  olatConfigurationKey: string
+  isSubselectionRequired: boolean
+  isEmailTransferRequired: boolean
+}
+
+interface ActivityTypeOfCourse {
+  id: string
+  title: string
+  olatConfigurationKey: string
+  isSubselectionRequired: boolean
+}
+
+let activityTypesAvailable: ActivityType[] = []
+const activityKeysSubselection: string[] = [
+  'live-quiz',
+  'practice-quiz',
+  'micro-learning',
+] as const // NOTE: add more if required
+let activityKeysGeneral: string[] = []
+type ActivityTypeSubselection = (typeof activityKeysSubselection)[number]
+
+async function apiKeyMiddleware(req: Request, res: Response, next: () => void) {
+  const apiKey = req.headers['x-api-key']
+  if (!apiKey || typeof apiKey !== 'string') {
+    return res.status(StatusCode.BAD_REQUEST).json({ error: 'Missing API key' })
+  }
+  if (apiKey !== API_KEY) {
+    return res
+      .status(StatusCode.UNAUTHORIZED)
+      .json({ error: 'Invalid API key' })
+  }
+  next()
+}
+
+const limiter = rateLimit({
+  windowMs: 60 * 1000, // 1 minute
+  max: 100,
+  standardHeaders: true, // RateLimit-* headers
+  legacyHeaders: false, // Disable `X-RateLimit-*` headers
+  message: {
+    error: 'Too many requests, please try again later.',
+  },
+})
+
+app.use(limiter)
+app.use('/api', apiKeyMiddleware)
+app.use(express.json())
+
+app.get('/health', (req: Request, res: Response) => {
+  res.status(StatusCode.SUCCESS).json({
+    status: 'OK',
+    timestamp: new Date().toISOString(),
+  })
+})
+
+app.get('/', (req: Request, res: Response) => {
+  res.json({ message: 'OLAT API server' })
+})
+
+// Serve OpenAPI specification
+app.get('/openapi.yaml', async (req: Request, res: Response) => {
+  try {
+    const __filename = fileURLToPath(import.meta.url)
+    const __dirname = path.dirname(__filename)
+    const specPath = path.join(__dirname, '../static/openapi.yaml')
+    const yamlContent = await fs.readFile(specPath, 'utf-8')
+    res.set('Content-Type', 'application/yaml')
+    res.send(yamlContent)
+  } catch (error) {
+    console.error('Error serving OpenAPI specification:', error)
+    res.status(StatusCode.INTERNAL_SERVER_ERROR).json({
+      error: 'Failed to load OpenAPI specification',
+    })
+  }
+})
+
+// Serve Scalar API documentation
+app.use(
+  '/api-docs',
+  apiReference({
+    url: '/openapi.yaml',
+    theme: 'default',
+  })
+)
+
+function checkHeader(req: Request): StatusCode {
+  const contentType = req.headers['content-type']
+  if (!contentType || typeof contentType !== 'string') {
```

Comment:
not sure we actually care about the content type so much that we should fail the request so strictly. I would just default to returning JSON, we don't support anything else.

so I would fail with bad request if anything other than application/json is defined, but otherwise always return json (also if content type is undefined)